gorillaz.go

```diff
@@ -246,13 +248,24 @@ func (g *Gaz) Run() <-chan struct{} {
 		err = g.httpSrv.Serve(httpListener)
 		if err != nil {
 			if err != http.ErrServerClosed {
-				Sugar.Errorf("HTTP server server stopped on :%d, %v", httpPort, err)
-				panic(err)
+				Log.Panic("HTTP serve stopped unexpectedly", zap.Error(err))
 			} else {
 				Sugar.Infof("HTTP server server stopped on :%d", httpPort)
 			}
 		}
 	}()
+	if g.ServiceDiscovery != nil {
+		Log.Info("registering service", zap.String("serviceName", g.ServiceName), zap.String("serviceAddr", g.serviceAddress), zap.Int("grpcPort", g.GrpcPort()), zap.Int("httpPort", g.HttpPort()))
+		var err error
+		g.registrationHandler, err = g.Register(&ServiceDefinition{ServiceName: g.ServiceName,
```

Comment:
Good idea.

Reply:
Currently with the consul implementation, the registration is done asynchronously, and even if it is not successful right away (e.g. the consul agent might be down), it will keep registering.
So in this context what would a timeout mean?